syntax = "proto3";

// Enum to represent the type of message being sent/received
enum MessageType {
    CREATE_PAYMENT = 0;
    UPDATE_PAYMENT = 1;
    PAYMENT_STATUS_UPDATE = 3;
}

enum PaymentStatus {
    PENDING = 0;
    COMPLETED = 1;
    FAILED = 2;
    REFUNDED = 3;
}

message Payment {
    int32 id = 1;
    int32 order_id = 2;
    int32 user_id = 3;
    float amount = 4;
    string currency = 5;
    string payment_method = 6;
    PaymentStatus status = 7;
    string created_at = 8;
    string updated_at = 9;
}


message PaymentMessage {
    MessageType message_type = 1;  
    Payment payment_data = 2; 
    int32 payment_id=3 ; //del     
}
